### PR TITLE
Hide auto-contribute section in brave://settings when region is JP

### PR DESCRIPTION
**Pull request by emerick**
Resolves https://github.com/brave/brave-browser/issues/19355

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### JP Region

- Clean profile
- Run brave with `--rewards=countryid=19024` option to simulate running in JP region
- Visit brave://settings
- Click on Rewards
- Click link in right pane to open Rewards panel
- Enable Rewards
- Verify that auto-contribution settings aren't shown in settings

### Non-JP Region

- Clean profile
- Visit brave://settings
- Click on Rewards
- Click link in right pane to open Rewards panel
- Enable Rewards
- Verify that auto-contribution settings are shown in settings

### CODE REVIEW

File: browser/extensions/api/brave_rewards_api.cc [hunk 1]
```diff
@@ -1058,6 +1058,23 @@ void BraveRewardsGetExternalWalletFunction::OnGetExternalWallet(
...
+  Profile* profile = Profile::FromBrowserContext(browser_context());
+  RewardsService* rewards_service =
+      RewardsServiceFactory::GetForProfile(profile);
+  if (!rewards_service) {
+    return RespondNow(Error("Rewards service is not initialized"));
+  }
+
+  const std::string external_wallet_type =
+      rewards_service->GetExternalWalletType();
+  return RespondNow(OneArgument(base::Value(external_wallet_type)));
```
Comment (szilardszaloki) [inline browser/extensions/api/brave_rewards_api.cc:1075]: Nit:
```
std::string external_wallet_type = rewards_service->GetExternalWalletType();
return RespondNow(OneArgument(base::Value(std::move(external_wallet_type))));
```
or even:
```
return RespondNow(OneArgument(base::Value(rewards_service->GetExternalWalletType())));
```

Comment (emerick) [reply browser/extensions/api/brave_rewards_api.cc:1075]: Good idea, I went with your second suggestion!

**Review by zenparsing — COMMENTED** [review]
In this PR, "AC supported" is derived from the "current external wallet type". Although this is status quo, we might want to think about moving away from that and just surfacing "AC supported" as its own thing from the rewards service.

The primary advantage, to me, would be that we could get rid of the idea of "current external wallet type", which isn't really meaningful in the context of an unverified user and breaks down when there are more than one supported external wallet providers.

**Review by zenparsing — APPROVED** [review]

**Comment by emerick** [comment]
Restarting Mac CI (all other platforms passed).

**Comment by emerick** [comment]
All CI passed, ready to merge.

**Comment by stephendonner** [comment]
Verified `PASSED` using

Brave | 1.34.20 Chromium: 96.0.4664.45 (Official Build) nightly (64-bit)
-- | --
Revision | 76e4c1bb2ab4671b8beba3444e61c0f17584b2fc-refs/branch-heads/4664@{#947}
OS | Windows 10 Version 20H2 (Build 19042.1348)

## JP region

1. clean profile
2. launched Brave using `--rewards-countryid=19024` to simulate running in JP region
3. opened `brave://settings`
4. clicked on `Brave Rewards` in the left pane
5. clicked the link in the right pane to open the `Rewards` panel
6. clicked on `Start using Brave Rewards`
7. clicked `Skip for now`
8. clicked `Continue`
9. clicked `Do it later`
10. dismissed the panel
11. looked at `brave://settings/rewards`

Confirmed there was no `Auto-Contribution` section under `Brave Rewards`
Additionally, confirmed no A-C on `brave://rewards` or the tips panel

example | example | example | example
----------|----------|-----------|----------
 |  |  | 




## Non-JP region

1. clean profile
2. launched Brave in `en-US` region
3. opened `brave://settings`
4. clicked on `Brave Rewards` in the left pane
5. clicked the link in the right pane to open the `Rewards` panel
6. clicked on `Start using Brave Rewards`
7. clicked `Skip for now`
8. clicked `Continue`
9. clicked `Do it later`
10. dismissed the panel
11. looked at `brave://settings/rewards`

Confirmed there was an `Auto-Contribution` section under `Brave Rewards`
Additionally, confirmed an auto-contribution was scheduled for `duckduckgo.com`

example | example | example | example | example
----------|----------|------------|----------|----------
 |  |  |  |

**Pull request merged**